### ISSUE TITLE
linux/build-static.sh is obsolete

### ISSUE THREAD

**Issue by cfsmp3** [CCExtractor/ccextractor#1348]
It builds CCExtractor 0.82...  we should either try to update it (best) or remove it.

**Comment by canihavesomecoffee** [comment]
It clones from master, so it will build the most recent... However the deps might not be up to date at all.

**Comment by cfsmp3** [comment]
> It clones from master, so it will build the most recent... However the deps might not be up to date at all.

Ah yes. We just like to leave irrelevant comments around
```
# wget https://sourceforge.net/projects/ccextractor/files/ccextractor/0.82/ccextractor.src.0.82.zip;
```

**Comment by cfsmp3** [comment]
@prateekmedia could you check this out? Probably you can fix it without a lot of pain.

Better - just make -static an arg to build :-)

**Comment by prateekmedia** [comment]
Will look at it

**Comment by prateekmedia** [comment]
Will look at it

**Comment by prateekmedia** [comment]
@cfsmp3 Could you explain what needs to be done in this? I looked at the file, ran it and it gave ccextractor-README.txt not found error.

**Comment by cfsmp3** [comment]
> @cfsmp3 Could you explain what needs to be done in this? I looked at the file, ran it and it gave `ccextractor-README.txt` not found error.

This will give you some context

https://github.com/CCExtractor/ccextractor/issues/597

Note that I renamed the script from @traceypooh to build-static.sh without actually changing the context (not a great idea, I realize).

Still, the script is obsolete now but the idea is good. 

Are we able to produce a static binary using Docker as container? (docker is great for this because you can have a reliable environment no matter the starting point in the host).

If you think that's not possible, or really a lot of work, I guess we should just remove build-static.sh for now because obviously it's not working and it's just going to confuse people.

**Comment by prateekmedia** [comment]
I will try to do some testing from my side, let's see if I get it working.

**Comment by prateekmedia** [comment]
@cfsmp3 I think building static is not something we should do, instead why not build an AppImage (which is a universal format) and it will be able to run without being dependent on the system dependencies.

**Comment by cfsmp3** [comment]
> @cfsmp3 I think building static is not something we should do, instead why not build an AppImage (which is a universal format) and it will be able to run without being dependent on the system dependencies.

That's a really good idea.
It would also solve the eternal issues with tesseract training files, etc

**Comment by prateekmedia** [comment]
@cfsmp3 Do you mean that tessdata should not be in the AppImage itself but installed separately?

**Comment by cfsmp3** [comment]
> @cfsmp3 Do you mean that tessdata should not be in the AppImage itself but installed separately?

No, the opposite! If we could have a AppImage that included tessdata it would be fantastic.

And if the image came from the CI so we could always have a known good nightly it would solve most of our issues with fixes taking forever to reach users because we release a new version once a year or so.

**Comment by IshanGrover2004** [comment]
Hey, Can i try making AppImage in replace of build-static.sh file
Or this issue is just for the CCExtracter team?

**Comment by cfsmp3** [comment]
You can work on it

**Comment by IshanGrover2004** [comment]
> You can work on it

Okk then
I'll research and will open PR after working on it

Any information I should know?

**Comment by prateekmedia** [comment]
@IshanGrover2004 You can began by reading appimage creation documentation and look at how other softwares are packaging the appimage.

**Comment by IshanGrover2004** [comment]
Hi,
I have studied about AppImage from its official docs and have gained a good understanding of how it is created.
I am planning to make a `.sh` file which will setup the `AppDir` and place all necessary binaries to make final `.AppImage` file and delete binaries & AppDir. I want to confirm if my approach is correct.

Could you please confirm if `build_static.sh` is complete reference file for this purpose, or if there are additional considerations I should keep in mind?

**Comment by IshanGrover2004** [comment]
I have opened PR #1592 . Can anyone from CCExtractor team look on it, so that i resolve current issues in there.
